Gems/ROS2/Code/Source/Camera/CameraSensor.cpp

```diff
@@ -45,58 +45,65 @@ namespace ROS2
-            localViewToClipMatrix, AZ::DegToRad(m_cameraConfiguration.m_verticalFieldOfViewDeg), m_aspectRatio, nearDist, farDist, true);
-        return localViewToClipMatrix;
-    }
+        //! Create a header-less CameraImage message.
+        sensor_msgs::msg::Image CreateImageMessageFromReadBackResult(
+            const AZ::EntityID& entityId, const AZ::RPI::AttachmentReadback::ReadbackResult& result, const std_msgs::msg::Header& header)
+        {
+            const AZ::RHI::ImageDescriptor& descriptor = result.m_imageDescriptor;
+            const auto format = descriptor.m_format;
+            AZ_Assert(Internal::FormatMappings.contains(format), "Unknown format in result %u", static_cast<uint32_t>(format));
+            sensor_msgs::msg::Image message;
+            message.encoding = Internal::FormatMappings.at(format);
+            message.width = descriptor.m_size.m_width;
+            message.height = descriptor.m_size.m_height;
+            message.step = message.width * Internal::BitDepth.at(format);
+            message.data = std::vector<uint8_t>(result.m_dataBuffer->data(), result.m_dataBuffer->data() + result.m_dataBuffer->size());
+            message.header = header;
+            bool registeredPostProcessingSupportsEncoding = false;
+            CameraPostProcessingRequestBus::EventResult(
+                registeredPostProcessingSupportsEncoding,
+                entityId,
+                &CameraPostProcessingRequests::SupportsFormat,
+                AZStd::string(Internal::FormatMappings.at(format)));
+            if (registeredPostProcessingSupportsEncoding)
+            {
+                CameraPostProcessingRequestBus::Event(entityId, &CameraPostProcessingRequests::ApplyPostProcessing, message);
+            }
+            return message;
+        }
 
-    void CameraSensorDescription::ValidateParameters() const
-    {
-        AZ_Assert(
-            m_cameraConfiguration.m_verticalFieldOfViewDeg > 0.0f && m_cameraConfiguration.m_verticalFieldOfViewDeg < 180.0f,
-            "Vertical fov should be in range 0.0 < FoV < 180.0 degrees");
-        AZ_Assert(!m_cameraName.empty(), "Camera name cannot be empty");
-    }
+        //! Prepare a header-less CameraInfo message.
+        sensor_msgs::msg::CameraInfo CreateCameraInfoMessage(
+            const CameraSensorDescription& cameraDescription, const std_msgs::msg::Header& header)
+        {
+            const auto& cameraIntrinsics = cameraDescription.m_cameraIntrinsics;
+            sensor_msgs::msg::CameraInfo cameraInfo;
+            cameraInfo.width = cameraDescription.m_cameraConfiguration.m_width;
+            cameraInfo.height = cameraDescription.m_cameraConfiguration.m_height;
+            cameraInfo.distortion_model = sensor_msgs::distortion_models::PLUMB_BOB;
+            AZ_Assert(cameraIntrinsics.size() == 9, "camera matrix should have 9 elements");
+            AZ_Assert(cameraInfo.k.size() == 9, "camera matrix should have 9 elements");
```

Comment:
Can we make `9` a constant?
```suggestion
            [[maybe_unused]] constexpr size_t expectedMatrixSize = 9;
            AZ_Assert(cameraIntrinsics.size() == expectedMatrixSize, "camera matrix should have %d elements", expectedMatrixSize);
            AZ_Assert(cameraInfo.k.size() == expectedMatrixSize, "camera matrix should have %d elements", expectedMatrixSize);
```